doc/sphinx/proof-engine/ltac.rst

```diff
@@ -713,339 +962,563 @@ tactic
-   is obtained by concatenating the value of the :n:`@component`\ s (each of them
-   is, either a :n:`@qualid` which has to refer to a (unqualified) name, or
-   directly a name denoted by a :n:`@string`).
+   Evaluates to an identifier unbound in the goal. This fresh identifier
+   is formed by concatenating the value of each :n:`@qualid`,
+   which has to refer to an unqualified name, and :n:`string`.
 
    If the resulting name is already used, it is padded with a number so that it
    becomes fresh. If no component is given, the name is a fresh derivative of
    the name ``H``.
 
-Computing in a constr
-~~~~~~~~~~~~~~~~~~~~~
+   :tacn:`fresh` is a :token:`value_tactic`.
+
+Computing in a term: eval
+~~~~~~~~~~~~~~~~~~~~~~~~~
 
 Evaluation of a term can be performed with:
 
 .. tacn:: eval @red_expr in @term
 
-   where :n:`@red_expr` is a reduction tactic among :tacn:`red`, :tacn:`hnf`,
-   :tacn:`compute`, :tacn:`simpl`, :tacn:`cbv`, :tacn:`lazy`, :tacn:`unfold`,
-   :tacn:`fold`, :tacn:`pattern`.
+   :tacn:`eval` is a :token:`value_tactic`.
 
-Recovering the type of a term
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Getting the type of a term
+~~~~~~~~~~~~~~~~~~~~~~~~~~
 
 .. tacn:: type of @term
 
    This tactic returns the type of :token:`term`.
 
-Manipulating untyped terms
-~~~~~~~~~~~~~~~~~~~~~~~~~~
+   :tacn:`type of` is a :token:`value_tactic`.
 
-.. tacn:: uconstr : @term
+Manipulating untyped terms: type_term
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-   The terms built in |Ltac| are well-typed by default. It may not be
-   appropriate for building large terms using a recursive |Ltac| function: the
-   term has to be entirely type checked at each step, resulting in potentially
-   very slow behavior. It is possible to build untyped terms using |Ltac| with
-   the :n:`uconstr : @term` syntax.
+The :n:`uconstr : ( @term )` construct can be used to build an untyped term.
+See :token:`syn_value`.
 
-.. tacn:: type_term @term
+.. tacn:: type_term @one_term
 
-   An untyped term, in |Ltac|, can contain references to hypotheses or to
+   In |Ltac|, an untyped term can contain references to hypotheses or to
    |Ltac| variables containing typed or untyped terms. An untyped term can be
-   type checked using the function type_term whose argument is parsed as an
+   type checked with :tacn:`type_term` whose argument is parsed as an
    untyped term and returns a well-typed term which can be used in tactics.
 
-Untyped terms built using :n:`uconstr :` can also be used as arguments to the
-:tacn:`refine` tactic. In that case the untyped term is type
+   :tacn:`type_term` is a :token:`value_tactic`.
+
+Untyped terms built using :n:`uconstr :` can be used as arguments to the
+:tacn:`refine` tactic, for example. In that case the untyped term is type
 checked against the conclusion of the goal, and the holes which are not solved
 by the typing procedure are turned into new subgoals.
```

Comment:
I think this example should be moved next to the description of `uconstr:` and does not belong with the description of `type_term`.